### DISCUSSION TITLE
Material

### DISCUSSION THREAD

**Discussion by MingSongJiang** [InfinyTech3D/SofaUnity#126] [Help - Using SofaUnity] [answered]
Hello,
I used gmsh to create a new.msh file for the cube. I generated the cube in unity, but when I added the material to the cube, the texture of the material didn't show up correctly.
Thanks!
![image](https://github.com/user-attachments/assets/cb631d18-9ebc-46da-b2ff-620b531cd007)

**Comment by epernod** [comment]
Hi,

are you sure about the uv texture?
Normally we use an obj file in addition to the msh file to handle the uv texture.

You can also try to change the texture coordinate system in the inspector of the Sofa Visual Model. See picture:
![image](https://github.com/user-attachments/assets/6b069659-2792-473a-8a12-cdba5cb9a6a2)

Best regards

**Reply by MingSongJiang** [reply]
Hi,
I tried changing the UV umwrap method parameter, but it didn't work.
The texture map of the cube created by Unity is uniform, while the texture map of the cube created by Gmsh is slanted towards the middle.The material is a new standard material created by Unity.



Thanks!
![image](https://github.com/user-attachments/assets/6d20eec9-00f7-4651-954d-ccdeae4afabb)

**Reply by MingSongJiang** [reply, marked as answer]
Just make the length and width of the cuboid the same.
